scala/private/rules/scala_library.bzl

```diff
@@ -251,3 +252,55 @@ def make_scala_macro_library(*extras):
     )
 
 scala_macro_library = make_scala_macro_library()
+
+
+##
+# scalajs_library
+##
+
+def _scalajs_library_impl(ctx):
+    return run_phases(
+        ctx,
+        # customizable phases
+        [
+            ("scalac_provider", phase_scalac_provider),
+            ("collect_srcjars", phase_collect_srcjars),
+            ("write_manifest", phase_write_manifest),
+            ("dependency", phase_dependency_common),
+            ("collect_jars", phase_collect_jars_common),
+            ("compile", phase_compile_scalajs_library),
```

Comment:
i think that there is a way to change the compile phase here w/ creating a whole new rule